SPECS/kubernetes/kubernetes-1.17.spec

```diff
@@ -0,0 +1,207 @@
+%global debug_package %{nil}
+%ifarch x86_64
+%define archname amd64
+%endif
+%ifarch aarch64
+%define archname arm64
+%endif
+%define kubeproxy_n_1 1.16.13
+%define coredns_n_1 1.6.2
+Summary:        Microsoft Kubernetes
+Name:           kubernetes
+Version:        1.17.11
+Release:        3%{?dist}
+License:        ASL 2.0
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Microsoft Kubernetes
+URL:            https://mcr.microsoft.com/oss
+#               Download URL https://kubernetesartifacts.azureedge.net/kubernetes/v1.17.11-hotfix.20200901/binaries/kubernetes-node-linux-amd64.tar.gz
+#               Note that only amd64 tarball exist which is OK since kubernetes is built from source
+Source0:        kubernetes-node-linux-amd64-%{version}-hotfix.20200901.tar.gz
+Source1:        kubelet.service
+BuildRequires:  golang >= 1.13.15
+BuildRequires:  rsync
+BuildRequires:  which
+BuildRequires:  flex-devel
+BuildRequires:  systemd-devel
+Requires:       cni
+Requires:       cri-tools
+Requires:       ebtables
+Requires:       ethtool
+Requires:       iproute
+Requires:       iptables
+Requires:       moby-engine
+Requires:       socat
+Requires:       util-linux
+Requires(postun): %{_sbindir}/groupdel
+Requires(postun): %{_sbindir}/userdel
+Requires(pre):  %{_sbindir}/groupadd
+Requires(pre):  %{_sbindir}/useradd
+
+%description
+Microsoft Kubernetes %{version}.
+
+%package        client
+Summary:        Client utilities
+
+%description    client
+Client utilities for Microsoft Kubernetes %{version}.
+
+%package        kubeadm
+Summary:        Bootstrap utilities
+Requires:       %{name} = %{version}
+Requires:       moby-cli
+
+%description    kubeadm
+Bootstrap utilities for Microsoft Kubernetes %{version}.
+
+%prep
+%setup -q -D -T -b 0 -n %{name}
+
+# note: kubernetes RPM can be build from binaries provided in source0 tarball
+#       by doing nothing in %build and %check sections
+%build
+# expand kubernetes source tarball (which is included source0 tarball)
+echo "+++ extract sources from tarball"
+mkdir -p %{_builddir}/%{name}/src
+cd %{_builddir}/%{name}/src
+tar -xof %{_builddir}/%{name}/kubernetes-src.tar.gz
+
+# build and update kubernetes components that are provided as binary
+# (other/unused kubernetes componenents will not be built)
+components_to_build=$(ls -1 %{_builddir}/%{name}/node/bin)
```

Comment:
what does components_to_build evaluates to? We should not be building apiserver and such here imho.

Reply:
this variable contain the list of the kubernetes binaries found in node/bin in kubernetes-node-linux-amd64.tar.gz tarball (source0).
The goal is to only rebuild the binaries that the upstream provide in the tarball